[Feat] 사용자 카테고리 및 목표 설정 기능 추가

ehddms0320 opened the pull request:
## ✅ PR 유형
어떤 변경 사항이 있었나요?

- [ ] 새로운 기능 추가

---

## ✏️ 작업 내용
- 로그인 사용자에 한해 초기 카테고리 및 목표를 설정할 수 있는 API를 구현함
- Category 조회 및 MissionSelection 생성/갱신 로직을 서비스 계층에 추가함
- 중복 설정 시 기존 MissionSelection을 갱신하도록 처리함

---

## 🔗 관련 이슈
- mission selection entity에 insertable=false, updatable=false 제거하였습니다 (JPA가 FK 컬럼 값을 넣을 수 있게)

---

## 💡 추가 사항

ehddms0320 committed:
[Feat] 사용자 카테고리 및 목표 설정 기능 추가
- 로그인 사용자에 한해 초기 카테고리 및 목표를 설정할 수 있는 API를 구현함
- Category 조회 및 MissionSelection 생성/갱신 로직을 서비스 계층에 추가함
- 중복 설정 시 기존 MissionSelection을 갱신하도록 처리함

Choe-Jong-Geon commented on src/main/java/umc9th_hackathon/daybreak/domain/member/controller/UserSetupController.java:
```diff
@@ -0,0 +1,31 @@
+
+    @PostMapping("/setup")
+    public ApiResponse<?> setup(
+            @RequestHeader("X-MEMBER-ID") Long memberId,
```
로그인하면 나오는 토큰값으로 사용자를 구별하게 해야할거 같아

Choe-Jong-Geon commented on src/main/java/umc9th_hackathon/daybreak/domain/member/service/UserSetupService.java:
```diff
@@ -0,0 +1,40 @@
+
+    public void setup(Long memberId, UserSetupRequest req) {
+        Member member = memberRepository.findById(memberId)
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 회원입니다."));
```
사용자를 찾을 수 없을 때 에러를 커스텀으로 만들어놔서 예외처리를 그걸로 반영했으면 좋겠어

ehddms0320 committed:
[Fix] 토큰 기반 사용자 식별로 초기 설정 API 수정
- X-MEMBER-ID 헤더 의존 제거
- 인증 사용자 기준으로 member 조회/처리

ehddms0320 committed:
[Fix] 사용자 설정 시 회원 미존재 예외 처리 개선
- 회원 조회 실패 시 IllegalArgumentException 대신 GeneralException(MemberErrorCode) 사용
- 전역 예외 처리 규약(ApiResponse)과 일관성 유지

Sharon0320 committed:
Merge branch 'develop' into feat/user_category_setup

Sharon0320 committed:
Merge branch 'develop' into feat/user_category_setup
